File: app/src/main/java/mx/edu/upqroo/kristenandroid/adapters/source/NoticeDataSourceFactory.java
package mx.edu.upqroo.kristenandroid.adapters.source;

import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;
import androidx.paging.PageKeyedDataSource;
import mx.edu.upqroo.kristenandroid.data.database.entities.Notice;

public class NoticeDataSourceFactory extends DataSource.Factory {

    private MutableLiveData<PageKeyedDataSource<Integer, Notice>> itemLiveDataSource = new MutableLiveData<>();

    @Override
    public DataSource<Integer, Notice> create() {
        //getting our data source object
        NoticeDataSource itemDataSource = new NoticeDataSource();

        //posting the data source to get the values
        itemLiveDataSource.postValue(itemDataSource);

        //returning the datasource
        return itemDataSource;
    }

    public MutableLiveData<PageKeyedDataSource<Integer, Notice>> getLiveDataSource() {
        return itemLiveDataSource;
    }
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/adapters/source/NoticeDataSourceFactory.kt
package mx.edu.upqroo.kristenandroid.adapters.source

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import androidx.paging.PageKeyedDataSource
import mx.edu.upqroo.kristenandroid.data.database.entities.Notice

class NoticeDataSourceFactory : DataSource.Factory<Int, Notice>() {

    val liveDataSource = MutableLiveData<PageKeyedDataSource<Int, Notice>>()

    override fun create(): DataSource<Int, Notice> {
        //getting our data source object
        val itemDataSource = NoticeDataSource()

        //posting the data source to get the values
        liveDataSource.postValue(itemDataSource)

        //returning the datasource
        return itemDataSource
    }
}
